onnxruntime/test/onnx/TestCase.cc

```diff
@@ -473,57 +494,105 @@ void OnnxTestCase::LoadTestData(size_t id, onnxruntime::test::HeapBuffer& b,
+      ConvertTestData(test_pb, b, is_input, i, name_data_map);
+    } else {
+      ORT_THROW("Unsupported type for the ", is_input ? "input " : "output ", i, " in the test runner");
+    }
+  }
 }
 
-void OnnxTestCase::ConvertTestData(const std::vector<ONNX_NAMESPACE::TensorProto>& test_data_pbs,
+void OnnxTestCase::ConvertTestData(const ONNX_NAMESPACE::TensorProto& test_data_pb,
                                    onnxruntime::test::HeapBuffer& b,
-                                   bool is_input, std::unordered_map<std::string, Ort::Value>& out) const {
-  bool has_valid_names = true;
-  std::vector<std::string> var_names(test_data_pbs.size());
-  for (size_t input_index = 0; input_index != test_data_pbs.size(); ++input_index) {
-    std::string name = test_data_pbs[input_index].name();
-    if (name.empty()) {
-      has_valid_names = false;
-      break;
-    }
-    var_names[input_index] = name;
+                                   bool is_input, size_t i,
+                                   std::unordered_map<std::string, Ort::Value>& out) const {
+  std::string name = test_data_pb.name();
+  std::string name_finalized = !name.empty() ? name : (is_input ? model_info_->GetInputName(i) : model_info_->GetOutputName(i));
+
+  size_t len = 0;
+
+  auto status = onnxruntime::test::GetSizeInBytesFromTensorProto<0>(test_data_pb, &len);
+  if (!status.IsOK()) {
+    ORT_THROW(status.ToString());
   }
-  if (!has_valid_names) {
-    size_t count = static_cast<size_t>(is_input ? model_info_->GetInputCount() : model_info_->GetOutputCount());
-    if (count != test_data_pbs.size()) {
-      ORT_THROW("data count mismatch, expect ", count, ", got ", test_data_pbs.size());
-    }
-    for (size_t i = 0; i != count; ++i) {
-      var_names[i] = is_input ? model_info_->GetInputName(i) : model_info_->GetOutputName(i);
-    }
+  void* p = len == 0 ? nullptr : b.AllocMemory(len);
+  Ort::Value v1{nullptr};
+  onnxruntime::test::OrtCallback d;
+  OrtMemoryInfo cpu_memory_info(onnxruntime::CPU, OrtDeviceAllocator, OrtDevice(), 0, OrtMemTypeDefault);
+  status = onnxruntime::test::TensorProtoToMLValue(test_data_pb, onnxruntime::test::MemBuffer(p, len, cpu_memory_info),
+                                                   v1, d);
+  if (!status.IsOK()) {
+    ORT_THROW(status.ToString());
+  }
+  if (d.f) {
+    b.AddDeleter(d);
   }
-  for (size_t input_index = 0; input_index != test_data_pbs.size(); ++input_index) {
-    std::string name = var_names[input_index];
-    const ONNX_NAMESPACE::TensorProto& input = test_data_pbs[input_index];
-    size_t len = 0;
+  out.emplace(name_finalized, std::move(v1));
+}
+
+void OnnxTestCase::ConvertTestData(const ONNX_NAMESPACE::SequenceProto& test_data_pb,
+                                   onnxruntime::test::HeapBuffer& b,
+                                   bool is_input, size_t i,
+                                   std::unordered_map<std::string, Ort::Value>& out) const {
+  std::string name = test_data_pb.name();
```

Comment:
nit: const ref

Reply:
Sure, but these diffs are misleading. This is the existing code as is. I can clean it up.

Reply:
Fixed